core/src/main/java/org/apache/accumulo/core/iterators/OrIterator.java

```diff
@@ -39,29 +39,44 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * An iterator that handles "OR" query constructs on the server side. This code has been adapted/merged from Heap and Multi Iterators.
+ * An iterator that provides a sorted-iteration of column qualifiers for a set of column families in a row.
+ * It is important to note that this iterator <em>does not</em> adhere to the contract set forth by the
+ * {@link SortedKeyValueIterator}. It returns Keys in {@code row+colqual} order instead of
+ * {@code row+colfam+colqual} order. This is required for the implementation of this iterator (to work
+ * in conjunction with the {@code IntersectingIterator}) but is a code-smell. This iterator should only
```

Comment:
> I don't think you have to insult the hygiene of our code unless you know of a use case where this design could cause problems

Is this tongue-in-cheek? I'm not sure if you think this comment is actually offensive to other developers.

It really is *bad*, IMO. "Apache Accumulo® is a sorted, distributed key/value store that..."  and we're providing an Iterator that, by all definitions, should be returning sorted data but is not. The intended use-case of this iterator is sitting behind the scenes (not feeding data directly to a Scanner/BatchScanner), which is why it's OK. I think it's important to let users know who venture here what they're getting into because it's unlike any (all?) other SKVI's that we provide.